numpy/lib/tests/test_function_base.py

```diff
@@ -2817,18 +2817,94 @@ def test_2D(self):
                       [1, 1, 1]])
         assert_array_equal(np.percentile(x, 50, axis=0), [1, 1, 1])
 
-    def test_linear(self):
-
+    TYPE_CODES = np.typecodes["AllInteger"] + np.typecodes["AllFloat"] + "O"
+
+    @pytest.mark.parametrize("dtype", TYPE_CODES)
+    def test_linear_inverted_cdf(self, dtype):
+        # METHOD 1 of H&F
+        arr = np.asarray([15.0, 20.0, 35.0, 40.0, 50.0], dtype=dtype)
```

Comment:
Could you apply this parametrization to all the newly added tests?

Reply:
oops I forgot that

Reply:
For information, when dtype is complex256 or float128 the result is not always exactly equal to the expected result.
For example with on `test_linear_median_unbiased` it gives `27.000000000000001776+0j)` instead of `27`.

Thus I changed the assertions to` np.testing.almost_equal` with either 14 or 15 precision to workaround this.